enhancements/single-node-developer-cluster-profile.md

```diff
@@ -0,0 +1,514 @@
+and expedite its implementation.
+
+Another risk is that it will take too long to get the initial
+annotations merged to all the needed OpenShift component repositories
+during the first phase of development. Until all needed annotations
+are merged, the SNC tooling cannot be updated to make use of the
+single-node-developer cluster profile, as needed components would be
+excluded from the cluster. The review process for this enhancement
+should help raise awareness across the various teams that will need to
+review and merge these patches.
+
+A final risk is that new components will be added to OpenShift after
+the completion of phase 1 that should be included in single node
+developer clusters, but will not have the necessary annotation
+included in their manifests due to lack of awareness or concern for
+the CRC product. Given that single node developer clusters are
+expected to pass the majority of the OpenShift conformance tests,
+component developers should include this profile's annotation unless
+the new component is clearly not relevant to the development of
+applications on single node clusters. Developer tools team members
+will be available to collaborate with component developers regarding
+whether a component should be included in the single-node-developer
+cluster profile, and if so, how it might be configured to minimize
+resource requirements.
+
+## Design Details
+
+### Open Questions
+
+One currently open question is how to manage the set of operators
+visible within the operator hub on a single-node-developer
+cluster. Operators that cannot run without OpenShift components
+excluded from the profile should not be available to install. Its not
+clear whether this is best achieved by producing a custom index for
+the OLM, having the OLM honor annotations in the manifests of
+operators it manages, or by dynamically evaluating whether
+dependencies are available in the cluster.
+
+### Test Plan
+
+Individual OpenShift components should not require any additional test
+cases during phase one, as the annotation for the single node
+developer cluster will be applied to existing manifest files that are
+already tested. Once these annotations are all merged and the SNC
+tooling is updated to have the OpenShift installer use the new cluster
+profile, the existing SNC tests and CRC product tests will verify that
+nothing has changed in the resulting cluster.
+
+Once phase one is complete, a new OpenShift CI job will be added to
+run the Kubernetes and OpenShift conformance test suites on a cluster
+deployed using the single-node-developer profile. Tests that require
+components disabled in the profile will be skipped. Note that these
+conformance tests are run without the additional changes made by the
+SNC image build tooling, so they do not necessarily reflect the actual
+conformance of the CRC product.
+
+As part of this CI job, or as a separate job, the memory utilization
```

Comment:
So in phase one you **only add annotations** and you magically expect that the memory utilization goes down? I thought that only after Phase 2 is completed you'll be able to achieve that lower footprint. In parallel, I'd suggest starting discussions with all the teams owning their components to include the necessary work for 4.8.

Reply:
> So in phase one you only add annotations and you magically expect that the memory utilization goes down? 
This paragraph says memory usage will be measured, so that we know the impact of the phase 2 changes, and so that we can detect if some components started using more memory during a development cycle.

Reply:
Not expecting any magic! If phase 1 does end up excluding manifests for any components that are included in the current CRC product, then we might see a small reduction in memory requirements from phase 1. Otherwise, the goal will be to measure and track this during phase two.